docs: Add arradio doc (#1604)
analogdevicesinc/hdl: docs/projects/arradio/index.rst

board by `Arrow <https://www.arrow.com/en/manufacturers/analog-devices>`__ & 
`Terasic <http://arradio.terasic.com/>`_ for the :adi:`AD9361`, a highly
integrated RF Agile Transceiver. The complete chip level design package can be
found on the ADI web site :adi:`ad9361_design_files`. Information on the card,
and how to use it, the design package that surrounds it, and the software
which can make it work, can be found on this page.

The ARRADIO board is not a product of Analog Devices, and questions about
purchase, or returns should go to Arrow. You can purchase the board from
`Arrow's web site <https://www.arrow.com/en/products/arradio/terasic-technologies>`__ .

The purpose of the ARRADIO board is to provide an RF platform to which shows
maximum performance of the :adi:`AD9361`. It's expected that the RF performance
of this platform can meet the datasheet specifications without issues at 2.4
GHz, and not much anywhere else. This is due to the external Johanson
Technology's `2450BL15B050E <https://www.johansontechnology.com/datasheets/baluns/JTI_Balun-2450BL15B050_12-03.pdf>`__ 
2.45 GHz Balun that is on the board. This balun is rated for a operating
frequency of 2400~2500 MHz.

> This platform is primarily for hardware/RF investigation and bring up of various
> waveforms from a RF team before their custom hardware is complete, where they
> want to see waveforms at their frequency of interest, and are not afraid of
> changing out the balun if necessary. (Have a look in the
> :dokuwiki:`Configuration </resources/eval/user-guides/ad-fmcomms2-ebz/hardware/configuration_options>`
> sections).

Reviewer: ```suggestion
This platform is primarily for hardware/RF investigation and bring-up of various
waveforms from a RF team before their custom hardware is complete. Take a look at the
:dokuwiki:`Configuration options<resources/eval/user-guides/ad-fmcomms2-ebz/hardware/configuration_options>`
section).
```

The ARRADIO board is very similar to the :adi:`AD-ARRADIO-EBZ`, except it utilizes
the HSMC connector which connects to the `Arrow SoCKit <https://www.arrow.com/en/products/sockit/arrow-development-tools>`__ .

Supported boards
-------------------------------------------------------------------------------

- `ARRADIO <https://www.terasic.com.tw/cgi-bin/page/archive.pl?No=946>`__

Supported devices
-------------------------------------------------------------------------------

- :adi:`AD9361`

Supported carriers
-------------------------------------------------------------------------------

>  - :intel:`C5SoC <content/www/us/en/products/details/fpga/development-kits/cyclone/v-sx.html>`
>    (Cyclone V SoC) on HSMC

Reviewer: ```suggestion
- :intel:`C5SoC <content/www/us/en/products/details/fpga/development-kits/cyclone/v-sx.html>`
  (Cyclone V SoC) on HSMC
```

     - OUT
     - 2
     - 2

Interrupts
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Below are the Programmable Logic interrupts used in the project.

=================== === =============== ================
Instance name       HDL Linux Cyclone V Actual Cyclone V
=================== === =============== ================
video_dmac          4   44              76
axi_ad9361_dac_dma  3   43              75
axi_ad9361_adc_dma  2   42              74
sys_spi             1   41              73
sys_gpio_bd         0   40              72
=================== === =============== ================

Building the HDL project
-------------------------------------------------------------------------------

The design is built upon ADI's generic HDL reference design framework.
ADI distributes the bit/elf files of these projects as part of the
:dokuwiki:`ADI Kuiper Linux <resources/tools-software/linux-software/kuiper-linux>`.
If you want to build the sources, ADI makes them available on the
:git-hdl:`HDL repository </>`. To get the source you must
`clone <https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository>`__
the HDL repository.

> Go to the hdl/projects/**arradio**/$carrier location and run the make

Reviewer: ```suggestion
Go to the hdl/projects/**arradio**/c5soc location and run the make
```

command.

**Linux/Cygwin/WSL**

.. shell::

   $cd hdl/projects/arradio/c5soc
   $make

A more comprehensive build guide can be found in the :ref:`build_hdl` user guide.

Resources
-------------------------------------------------------------------------------

Systems related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Here you can find the quick start guides available for these evaluation boards:

Reviewer: ```suggestion
Here you can find the quick start guides available for this evaluation board:
```

- :dokuwiki:`ARRADIO <resources/eval/user-guides/arradio>`
- :dokuwiki:`ARRADIO Terasic C5 SoCkit Quick Start Guide <resources/eval/user-guides/arradio/quickstart/alterasockit>`

Hardware related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> - `AD9361 reference manual <https://ez.analog.com/cfs-file/__key/telligent-evolution-components-attachments/00-441-00-00-00-07-91-97/AD9361_5F00_Reference_5F00_Manual_5F00_UG_2D00_570.pdf>`__

Reviewer: ```suggestion
- `AD9361 Reference Manual`_
```
